src/condor_shadow.V6.1/baseshadow.cpp

```diff
@@ -346,30 +346,25 @@ int BaseShadow::cdToIwd() {
 
 
 void
-BaseShadow::shutDownFast( int reason ) {
+BaseShadow::shutDownFast(int reason, const char* reason_str, int reason_code, int reason_subcode) {
 	m_force_fast_starter_shutdown = true;
-	shutDown( reason );
+	shutDown(reason, reason_str, reason_code, reason_subcode);
 }
 
 void
-BaseShadow::shutDown( int reason ) 
+BaseShadow::shutDown(int reason, const char* reason_str, int reason_code, int reason_subcode) 
 {
 		// exit now if there is no job ad
 	if ( !getJobAd() ) {
 		DC_Exit( reason );
 	}
 		//Attempt to write Job ad to epoch file
 		//If knob isn't set or there is no job ad the function will just log and return
-	writeJobEpochFile(getJobAd());
-		// if we are being called from the exception handler, return
-		// now to prevent infinite loop in case we call EXCEPT below.
-	if ( reason == JOB_EXCEPTION ) {
-		return;
-	}
 
```

Comment:
I don't see how the new code handles this case.

Reply:
I didn't see a way for the exception handler (or any code path using JOB_EXCEPTION) to end up here.

Reply:
Fair enough.  that code is from 2002, so maybe its no longer needed.